### ISSUE TITLE
Consider supporting .jinja2 extensions across the project

### ISSUE THREAD

**Issue by oripwk** [apache/airflow#8603]
**Description**

At the moment, various operators support reading templates from given filenames (for example, [bash_command](https://airflow.apache.org/docs/stable/_api/airflow/operators/bash_operator/index.html#airflow.operators.bash_operator.BashOperator) of `BashOperator`). However, all of these operators specify extensions such as `.sh`, .`json` etc., so it means users need to write Jinja2 templates inside `.sh`, `.json` files. A better way to do it would be supporting the `.jinja2` extension.

**Use case / motivation**

The motivation is threefold:
1. `.jinja2` extension is a widespread convention which is supported by various editors, including Github itself, PyCharm and [VSCode](https://marketplace.visualstudio.com/items?itemName=samuelcolvin.jinjahtml).
1. Putting templates inside `.jinja2` can help distinguish between regular files and template files just by looking at the directory contents
1. Working with tools that recognize `.jinja2` templates can help reduce errors while authoring templates.

**Related Issues**

https://github.com/apache/airflow/pull/8572#discussion_r416473395

**Comment by mik-laj** [comment]
Hi. 

It looks good. I think we can accept jinja2 whenever template ext is not empty.
Do you want to work on it? Should I assign you to this ticket?

Best regards,
Kaamil

**Comment by oripwk** [comment]
Hi,

Yes, I can work on it. You can assign

**Comment by potiuk** [comment]
Hey @oripwk. I generally like the idea, but can you also write about this one on the devlist - I think this is quite a fundamental change int the approach for potentially many operators, and I feel we should have more people commenting on it and maybe coming up with some ideas how to handle it well. I think for example that better  approach will be to simply support additional .jinja2 extension when you have extensions defined. For example when you have '.sh' extension in template extensions, you'd look in both `*.sh` and `*.sh.jinja2` but not just `*.jinja2`. This way we might preserve both - nice formatting support from IDE's and semantic information about the actual type of the file. But  this is just my opinion and maybe we can discuss it at the devlist.

Woudl you be so kind to write about the proposal at the devlist please ?

**Comment by oripwk** [comment]
Sure.
I also commented on the PR regarding how to generate documentation for this, since it affects many operators. Do you have any suggestion how to deal with it? https://github.com/apache/airflow/pull/8616#issuecomment-621086618

**Comment by eladkal** [comment]
@kaxil this issue should be resolved by https://github.com/apache/airflow/pull/14603 correct?

**Comment by kaxil** [comment]
> @kaxil this issue should be resolved by #14603 correct?

No, actually this is a bit different i.e. the OP wants to allow using "file.jinja2" instead of "file.bash" or "file.sql" in rendered template fields.

**Comment by uranusjr** [comment]
This would be irrelevant if [AIP-80](https://cwiki.apache.org/confluence/x/2grOEg) is accepted. Instead of inferring from the extension, you’d need to pass in a FileTemplate class explicitly, which can accept any file name.

**Comment by bugraoz93** [comment]
Are we going to make this happen? I suggest we move to 3.1+ otherwise :)

**Comment by uranusjr** [comment]
Yes this should be moved to 3.1.

**Comment by potiuk** [comment]
@oripwk We are unassigning you from this issue as part of our updated [assignment policy](https://github.com/apache/airflow/blob/main/contributing-docs/04_how_to_contribute.rst#contribute-code-changes).

This is not meant to discourage your contribution — quite the opposite! You are still very welcome to work on this issue and submit a PR for it. Simply comment that you are working on it and open a PR when ready.

We found that formal assignments were not working well, as they often prevented others from contributing when the assignee was not actively working on the issue.
